contracts/ERC827/proposals/ERC827TokenMockAllowedCallbacks.sol

```diff
@@ -14,42 +14,59 @@ import "../../ERC20/StandardToken.sol";
  */
 contract ERC827TokenAllowedCallbacks is StandardToken {
 
+  // @dev Enum to represent the type of token functions in uint8
+  enum FunctionType { None, Approve, Transfer, TransferFrom }
```

Comment:
Does it make sense to add another type `All`?

Reply:
I had it when I first implemented it, but I dont know how necessary it is, I think the callbacks allowance should be as restrictive as It can, I dont see the option "All" being used that much, and I dont know how safe it is to use it. I see approval as the most important one for callbacks and more secure, and this is the one that should be only used if you can.
It also make more complex the check for callbacks allowed (and this is executed in every ERC827 function).
If you really want to allow callback for all functions it will be calling allowCallbak function three times instead of one.
But again, I wont reccomend allowing callbacks for all functions at least you are 100% sure that is safe.